alignak/alignakobject.py

```diff
@@ -20,70 +20,161 @@
-            return
-        all_props = {}
-        all_props.update(getattr(self, "properties", {}))
-        all_props.update(getattr(self, "running_properties", {}))
+            params = {}
+
+        if debug:
+            print('AlignakObject __init__: %s, %d params' %
+                  (self.__class__, len(params)))
+            print('AlignakObject __init__: %s, params list: %s' %
+                  (self.__class__, [key for key in params]))
+            print('AlignakObject __init__: %s, %d properties' %
+                  (self.__class__, len(self.properties)))
+            print('AlignakObject __init__: %s, properties list: %s' %
+                  (self.__class__, [key for key in self.properties]))
+            print('AlignakObject __init__: %d running properties' %
+                  len(getattr(self, "running_properties", {})))
+            print('AlignakObject __init__: %s, running properties list: %s' %
+                  (self.__class__, [key for key in self.running_properties]))
+
+        # Fresh new object or object update/unserialization?
+        if parsing:
+            self.uuid = uuid.uuid4().hex
+
+        # Create attributes for class properties, running properties and macros (to be confirmed?)
+        object_properties = {}
+        object_properties.update(getattr(self.__class__, "properties", {}))
+        object_properties.update(getattr(self.__class__, "running_properties", {}))
+        # object_properties.update(getattr(self, "macros", {}))
+        index = 0
         for key, value in params.iteritems():
-            if key in all_props and isinstance(all_props[key], SetProp):
+            index = index + 1
+
+            if hasattr(self, key):
+                if debug:
+                    print('AlignakObject __init__: %s still exists (%s)' %
+                          (key, getattr(self, key)))
+                continue
+
+            if key in object_properties and isinstance(object_properties[key], SetProp):
                 setattr(self, key, set(value))
             else:
                 setattr(self, key, value)
 
-        if not hasattr(self, 'uuid'):
+            if debug:
+                print("%c %3d: %s = %s" %
+                      ('!' if key not in object_properties else ' ',
+                       index, key, getattr(self, key, 'not_set')))
+
+        if getattr(self, 'uuid', None) is None:
+            # Alert about missing object uuid
+            warnings.warn("Object do not have an uuid! %s" % self, RuntimeWarning, stacklevel=2)
+            # print("Problem with an object '%s', no uuid in: %s!" % (self.my_type, self.__dict__))
             self.uuid = uuid.uuid4().hex
 
-    def serialize(self):
-        """This function serialize into a simple dict object.
+        # Fill default values for unset running properties
+        # Default values for properties are set-up later in the initialization process...
+        self.fill_default(which_properties="running_properties")
```

Comment:
Yes, but for objects depend on AlignakObject directly and so not depend on Item, we can have problem